tests/Endpoint_selenium_unittest.py

```diff
@@ -0,0 +1,134 @@
+        # Assert ot the query to dtermine status of failure
+        self.assertTrue(re.search(r'Endpoint added successfully', productTxt))
+
+    def test_edit_endpoint(self):
+        # Login to the site. Password will have to be modified
+        # to match an admin password in your own container
+        driver = self.login_page()
+        # Navigate to the endpoint page
+        driver.get(self.base_url + "endpoint")
+        # Select one of the previously created endpoint to edit
+        driver.find_element_by_link_text("moving.com.rnd").click()
+        # "Click" the dropdown button to see options
+        driver.find_element_by_id("dropdownMenu1").click()
+        # "Click" the Edit Endpoint
+        driver.find_element_by_link_text("Edit Endpoint").click()
+        # Clear the old endpoint host name
+        driver.find_element_by_id("id_host").clear()
+        # Fill in the endpoint host name
+        driver.find_element_by_id("id_host").send_keys("https://moving.com/")
+        # Fill in port for endpoint
+        driver.find_element_by_id("id_port").clear()
+        driver.find_element_by_id("id_port").send_keys("8080")
+        # "Click" the submit button to complete the transaction
+        driver.find_element_by_css_selector("input.btn.btn-primary").click()
+        # Query the site to determine if the product has been added
+        productTxt = driver.find_element_by_tag_name("BODY").text
+        # Assert ot the query to dtermine status of failure
+        self.assertTrue(re.search(r'Endpoint updated successfully', productTxt))
+
+    def test_delete_endpoint(self):
+        # Login to the site. Password will have to be modified
+        # to match an admin password in your own container
+        driver = self.login_page()
+        # Navigate to the endpoint page
+        driver.get(self.base_url + "endpoint")
+        # Select one of the previously created endpoint to delete
+        driver.find_element_by_link_text("123.22.43.101").click()
+        # "Click" the dropdown button to see options
+        driver.find_element_by_id("dropdownMenu1").click()
+        # "Click" the Delete Endpoint
+        driver.find_element_by_link_text("Delete Endpoint").click()
+        # "Click" the delete button to complete the transaction
+        driver.find_element_by_css_selector("button.btn.btn-danger").click()
+        # Query the site to determine if the product has been added
+        productTxt = driver.find_element_by_tag_name("BODY").text
+        # Assert ot the query to dtermine status of failure
+        self.assertTrue(re.search(r'Endpoint and relationships removed.', productTxt))
+
+    def tearDown(self):
+        self.driver.quit()
+        self.assertEqual([], self.verificationErrors)
+
+
+def suite():
+    suite = unittest.TestSuite()
+    # Add each test the the suite to be run
+    # success and failure is output by the test
+    suite.addTest(product_unit_test.ProductTest('test_create_product'))
```

Comment:
Be sure to clean up any extra resources created. Deleting products also deletes any engagements/findings/.etc.

Reply:
yea.. taken care of